plugin.rb

```diff
@@ -292,26 +257,29 @@ module ::Jobs
       class VoteRelease < ::Jobs::Base
         def execute(args)
           if topic = Topic.with_deleted.find_by(id: args[:topic_id])
-            UserCustomField.where(name: DiscourseVoting::VOTES, value: args[:topic_id]).find_each do |user_field|
-              user = User.find(user_field.user_id)
-              user.custom_fields[DiscourseVoting::VOTES] = user.votes.dup - [args[:topic_id]]
-              user.custom_fields[DiscourseVoting::VOTES_ARCHIVE] = user.votes_archive.dup.push(args[:topic_id]).uniq
-              user.save!
-            end
+            votes = DiscourseVoting::Vote.where(topic_id: args[:topic_id])
+            votes.update_all(archive: true)
+
             topic.update_vote_count
+
+            return if args[:trashed]
+
+            votes.find_each do |vote|
+              Notification.create!(user_id: vote.user_id,
+                                   notification_type: Notification.types[:votes_released],
+                                   topic_id: vote.topic_id,
+                                   data: { message: "votes_released",
+                                           title: "votes_released" }.to_json)
```

Comment:
This has the chance to raise an exception so I think we need to either create the notifications in a transaction (all or nothing) or just rescue from the error and continue with the notification creation.